### PR TITLE
Add GetFileHash and VerifyFileHash tasks to XSD (for IntelliSense)

### PR DESCRIPTION
**Pull request by santoshyadav198613**

### CODE REVIEW

**Comment by santoshyadav198613** [comment]
Sample xml schema or reference:

<?xml version="1.0" encoding="utf-8"?>
<GetFileHash Files="anySimpleType" Algorithm="SHA256" MetadataName="FileHash" HashEncoding="hex" Hash="anySimpleType" Items="anySimpleType" Condition="Condition1" ContinueOnError="ContinueOnError1" Architecture="Architecture1" Runtime="Runtime1" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Output TaskParameter="TaskParameter1" ItemName="ItemName1" PropertyName="PropertyName1" Condition="Condition1" />
  <Output TaskParameter="TaskParameter2" ItemName="ItemName2" PropertyName="PropertyName2" Condition="Condition2" />
  <Output TaskParameter="TaskParameter3" ItemName="ItemName3" PropertyName="PropertyName3" Condition="Condition3" />
</GetFileHash>

<?xml version="1.0" encoding="utf-8"?>
<VerifyFileHash File="anySimpleType" Hash="anySimpleType" HashEncoding="hex" Algorithm="SHA256" Condition="Condition1" ContinueOnError="ContinueOnError1" Architecture="Architecture1" Runtime="Runtime1" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Output TaskParameter="TaskParameter1" ItemName="ItemName1" PropertyName="PropertyName1" Condition="Condition1" />
  <Output TaskParameter="TaskParameter2" ItemName="ItemName2" PropertyName="PropertyName2" Condition="Condition2" />
  <Output TaskParameter="TaskParameter3" ItemName="ItemName3" PropertyName="PropertyName3" Condition="Condition3" />
</VerifyFileHash>

**Comment by rainersigwald** [comment]
@santoshyadav198613 When you type XML in GitHub-flavored Markdown it often gets lost, but if you prefix it with `` ```xml`` and end it with `` ``` `` (making a syntax-highlighted [fenced code block](https://help.github.com/en/articles/creating-and-highlighting-code-blocks#fenced-code-blocks), it'll come out nice, like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<GetFileHash Files="anySimpleType" Algorithm="SHA256" MetadataName="FileHash" HashEncoding="hex" Hash="anySimpleType" Items="anySimpleType" Condition="Condition1" ContinueOnError="ContinueOnError1" Architecture="Architecture1" Runtime="Runtime1" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Output TaskParameter="TaskParameter1" ItemName="ItemName1" PropertyName="PropertyName1" Condition="Condition1" />
  <Output TaskParameter="TaskParameter2" ItemName="ItemName2" PropertyName="PropertyName2" Condition="Condition2" />
  <Output TaskParameter="TaskParameter3" ItemName="ItemName3" PropertyName="PropertyName3" Condition="Condition3" />
</GetFileHash>

<?xml version="1.0" encoding="utf-8"?>
<VerifyFileHash File="anySimpleType" Hash="anySimpleType" HashEncoding="hex" Algorithm="SHA256" Condition="Condition1" ContinueOnError="ContinueOnError1" Architecture="Architecture1" Runtime="Runtime1" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Output TaskParameter="TaskParameter1" ItemName="ItemName1" PropertyName="PropertyName1" Condition="Condition1" />
  <Output TaskParameter="TaskParameter2" ItemName="ItemName2" PropertyName="PropertyName2" Condition="Condition2" />
  <Output TaskParameter="TaskParameter3" ItemName="ItemName3" PropertyName="PropertyName3" Condition="Condition3" />
</VerifyFileHash>
```

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 1]
```diff
@@ -2397,6 +2397,20 @@ elementFormDefault="qualified">
...
+    <xs:element name="GetFileHash" substitutionGroup="msb:Task">
+        <xs:complexType>
+            <xs:complexContent>
+                <xs:extension base="msb:TaskType">
+                    <xs:attribute name="Files" use="required" />
+                    <xs:attribute name="Algorithm" default="SHA256" />
+                    <xs:attribute name="MetadataName" default="FileHash" />
+                    <xs:attribute name="HashEncoding" default="hex" />
+                    <xs:attribute name="Hash" />
+                    <xs:attribute name="Items" />
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: `Hash` and `Items` are outputs, so I don't think they should be listed here. But outputs are listed for other tasks. That confuses me. I guess leave it in.

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 2]
```diff
@@ -2397,6 +2397,20 @@ elementFormDefault="qualified">
             </xs:complexContent>
         </xs:complexType>
     </xs:element>
+    <xs:element name="GetFileHash" substitutionGroup="msb:Task">
+        <xs:complexType>
+            <xs:complexContent>
+                <xs:extension base="msb:TaskType">
+                    <xs:attribute name="Files" use="required" />
+                    <xs:attribute name="Algorithm" default="SHA256" />
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Is `default` meaningful here? I don't see it in use for any of the other tasks.

Comment (santoshyadav198613) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: I was reading about xsd and found this attribute, and in code also we have mentioned the default values, let me know if i should rollback.

Comment (rainersigwald) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: I'd love to have the extra information, but does it work in Visual Studio? Can you modify your local copy of the XSD and grab a screenshot, maybe?

Comment (santoshyadav198613) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: You are right not working , removing the default values.

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 1 above)
Comment (santoshyadav198613) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: ok

**Comment by santoshyadav198613** [comment]
Hi @rainersigwald ,

Thanks a lot for your help, done the changes.

**Pull request merged**